src/Factory/MenuFactory.php

```diff
@@ -44,13 +46,19 @@ public function __invoke(ContainerInterface $c)
             ->addLineBreak()
             ->addStaticItem('Exercises')
             ->addStaticItem('---------')
-            ->addItems(array_map(function (ExerciseInterface $exercise) use ($exerciseRenderer, $userState) {
-                return [
-                    $exercise->getName(),
-                    $exerciseRenderer,
-                    $userState->completedExercise($exercise->getName())
-                ];
-            }, $exerciseRepository->findAll()))
+            ->addItems(
+                array_map(function (ExerciseInterface $exercise) use ($exerciseRenderer, $userState, $workshopType) {
+                    $isCurrent  = $exercise->getName() === $userState->getCurrentExercise();
+                    $isComplete = in_array($exercise->getName(), $userState->getCompletedExercises());
+
+                    return [
+                        $exercise->getName(),
+                        $exerciseRenderer,
+                        $userState->completedExercise($exercise->getName()),
+                        $workshopType == WorkshopType::TUTORIAL() && !$isCurrent && !$isComplete
```

Comment:
Also can you extract this check to a method, it's not really obvious what it's doing